flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -528,8 +529,9 @@ public static void deleteExtractedLibraries(List<File> tempLibraries) {
 
 	private static void checkJarFile(URL jarfile) throws ProgramInvocationException {
```

Comment:
Don't we still this checkJarFile method ? It looks like we use call `JarUtils.checkJarFile` directly ?

Reply:
It wraps exception into `ProgramInvocationException`. I think whether or not refactor these codes can go into another story.